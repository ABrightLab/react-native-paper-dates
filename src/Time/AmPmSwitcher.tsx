import * as React from 'react'
import { View, StyleSheet } from 'react-native'
import { Text, TouchableRipple, useTheme } from 'react-native-paper'

export default function AmPmSwitcher() {
  const theme = useTheme()
  return (
    <View
      style={[
        styles.root,
        {
          borderColor: theme.colors.backdrop,
          borderRadius: theme.roundness,
        },
      ]}
    >
      <SwitchButton label="AM" onPress={() => {}} />
      <View
        style={[
          styles.switchSeparator,
          { backgroundColor: theme.colors.backdrop },
        ]}
      />
      <SwitchButton label="PM" onPress={() => {}} />
    </View>
  )
}

function SwitchButton({
  label,
  onPress,
}: {
  label: string
  onPress: () => any
}) {
  const theme = useTheme()
  return (
    <TouchableRipple onPress={onPress} style={styles.switchButton}>
      <Text selectable={false} style={{ ...theme.fonts.medium }}>
        {label}
      </Text>
    </TouchableRipple>
  )
}

const styles = StyleSheet.create({
  root: {
    width: 50,
    height: 96,
    borderWidth: 1,
  },
  switchSeparator: {
    height: 1,
    width: 41,
  },
  switchButton: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
})
