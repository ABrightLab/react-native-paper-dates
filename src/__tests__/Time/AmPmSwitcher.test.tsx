import * as React from 'react'

import renderer from 'react-test-renderer'
import AmPmSwitcher from '../../Time/AmPmSwitcher'

it('renders AmPmSwitcher', () => {
  const tree = renderer
    .create(<AmPmSwitcher hours={12} onChange={() => null} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
