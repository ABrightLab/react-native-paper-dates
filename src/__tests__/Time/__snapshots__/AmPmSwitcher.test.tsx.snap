// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AmPmSwitcher 1`] = `
<View
  style={
    [
      {
        "borderWidth": 1,
        "height": 80,
        "overflow": "hidden",
        "width": 50,
      },
      {
        "borderColor": "#E0D2E0",
        "borderRadius": 4,
      },
    ]
  }
>
  <View
    accessibilityComponentType="button"
    accessibilityLabel="AM"
    accessibilityRole="button"
    accessibilityState={
      {
        "disabled": true,
      }
    }
    accessibilityTraits={
      [
        "button",
        "disabled",
      ]
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        false,
        false,
        {
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          {
            "backgroundColor": "rgba(234, 221, 255, 1)",
          },
        ]
      }
    >
      <Text
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "color": "rgba(103, 80, 164, 1)",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "400",
                "letterSpacing": 0.25,
                "lineHeight": 20,
              },
            ],
          ]
        }
      >
        AM
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "height": 1,
          "width": 48,
        },
        {
          "backgroundColor": "#E0D2E0",
        },
      ]
    }
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel="PM"
    accessibilityRole="button"
    accessibilityState={
      {
        "disabled": false,
      }
    }
    accessibilityTraits="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        false,
        false,
        {
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          {
            "backgroundColor": "rgba(255, 251, 254, 1)",
          },
        ]
      }
    >
      <Text
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "color": "rgba(73, 69, 79, 1)",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "400",
                "letterSpacing": 0.25,
                "lineHeight": 20,
              },
            ],
          ]
        }
      >
        PM
      </Text>
    </View>
  </View>
</View>
`;
